internal/fs/fs.go

```diff
@@ -976,6 +976,16 @@ func (fs *fileSystem) lookUpOrCreateChildDirInode(
 func (fs *fileSystem) syncFile(
 	ctx context.Context,
 	f *inode.FileInode) (err error) {
+	// Do not sync local file if it has been deleted from the localFileInodes map.
+	fs.mu.Lock()
+	_, ok := fs.localFileInodes[f.Name()]
+	if f.IsLocal() && !ok {
+		// Do not return any error. This is in sync with non-local file behaviour.
```

Comment:
instead add a comment saying deletion from localFileInodes is done during unlink operation. SyncFile will be triggered for unlinked file if the fileHandle is open by another user/terminal. Hence ignoring the syncFile call.

Reply:
done